src/routes/mod.rs

```diff
@@ -51,6 +54,12 @@ pub fn active_routes() -> Vec<Route> {
         hooks::routes::flush
     ];
 
+    let preview_route = if is_preview_endpoint_enabled() {
```

Comment:
Personal preference:
I would initialize the array and push the endpoint depending on the feature flag, rather than having a conditional initialization. This way it is easier to add more endpoints that are secured by feature flags.

Reply:
Do you mean having a function that returns routes under feature flags and inside that function we check the feature flags one by one?